### DISCUSSION TITLE
is it possible to support a custom file extractor?

### DISCUSSION THREAD

**Discussion by 7801943** [langgenius/dify#11110] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

it seems like that the current doc extractor can't extract table content in .docx format

**Comment by mmnickou** [comment]
I tried to modify extract_processor.py in the api container, but the modification never applied to the web-api. Don't know if I missed anything.

**Comment by Yevanchen** [comment]
You can build a dify plugin yourself and take the array file as an input parameter, then define the file extraction and processing logic in the Python file
Reference: https://github.com/Yevanchen/Dify-Reag-Plugin

**Comment by 7801943** [comment]
thanks



发自我的iPhone


------------------ Original ------------------
From: Yevanchen ***@***.***&gt;
Date: Tue,Feb 11,2025 4:44 PM
To: langgenius/dify ***@***.***&gt;
Cc: 7801943 ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [langgenius/dify] is it possible to support a custom file extractor? (Discussion #11110)





 
You can build a dify plugin yourself and take the array file as an input parameter, then define the file extraction and processing logic in the Python file
 Reference: https://github.com/Yevanchen/Dify-Reag-Plugin
